packages/edit-site/src/components/global-styles/color-panel.js

```diff
@@ -82,12 +91,9 @@ export default function ColorPanel( {
 
 	let gradientSettings = {};
 	if ( hasGradientColor ) {
-		const gradient = getStyle( name, 'background' );
-		const userGradient = getStyle( name, 'background', 'user' );
 		gradientSettings = {
 			gradientValue: gradient,
-			onGradientChange: ( value ) =>
-				setStyle( name, 'background', value ),
+			onGradientChange: setGradient,
```

Comment:
Currently setting a background color or gradient is not working. That happens because for example when we set the background color, the panel calls setBackgroundColor followed by setGradient to clear a gradient that may exist. The problem is that both setBackgroundColor and setGradient calls operate on the same object so setGradient reverts the change made by setBackgroundColor. setGradient should operate on the structure changed by setBackgroundColor. We have a race condition happening on our hooks.

Reply:
I had the same issue at some point in the `color.js` block support, We'll see how I fixed that.

Reply:
should be fixed in 8858351